Source/Core/VideoCommon/Assets/TextureAsset.cpp

```diff
@@ -19,78 +20,120 @@ bool ParseSampler(const VideoCommon::CustomAssetLibrary::AssetID& asset_id,
-  Common::ToLower(&sampler_state_mode);
+    if (!sampler_state_wrap_iter->second.is<picojson::object>())
+    {
+      ERROR_LOG_FMT(VIDEO, "Asset '{}' failed to parse json, 'wrap_mode' is not the right type",
+                    asset_id);
+      return false;
+    }
+    const auto sampler_state_wrap_obj = sampler_state_wrap_iter->second.get<picojson::object>();
 
-  if (sampler_state_mode == "clamp")
-  {
-    sampler->tm0.wrap_u = WrapMode::Clamp;
-    sampler->tm0.wrap_v = WrapMode::Clamp;
-  }
-  else if (sampler_state_mode == "repeat")
-  {
-    sampler->tm0.wrap_u = WrapMode::Repeat;
-    sampler->tm0.wrap_v = WrapMode::Repeat;
-  }
-  else if (sampler_state_mode == "mirrored_repeat")
-  {
-    sampler->tm0.wrap_u = WrapMode::Mirror;
-    sampler->tm0.wrap_v = WrapMode::Mirror;
-  }
-  else
-  {
-    ERROR_LOG_FMT(VIDEO,
-                  "Asset '{}' failed to parse json, 'texture_mode' has an invalid "
-                  "value '{}'",
-                  asset_id, sampler_state_mode);
-    return false;
-  }
+    for (const auto& uv : std::array<std::string, 2>{"u", "v"})
+    {
+      auto uv_mode = ReadStringFromJson(sampler_state_wrap_obj, uv).value_or("");
+      Common::ToLower(&uv_mode);
 
-  const auto sampler_state_filter_iter = json.find("texture_filter");
-  if (sampler_state_filter_iter == json.end())
-  {
-    ERROR_LOG_FMT(VIDEO, "Asset '{}' failed to parse json, 'texture_filter' not found", asset_id);
-    return false;
-  }
-  if (!sampler_state_filter_iter->second.is<std::string>())
-  {
-    ERROR_LOG_FMT(VIDEO, "Asset '{}' failed to parse json, 'texture_filter' is not the right type",
-                  asset_id);
-    return false;
-  }
-  std::string sampler_state_filter = sampler_state_filter_iter->second.to_str();
-  Common::ToLower(&sampler_state_filter);
-  if (sampler_state_filter == "linear")
-  {
-    sampler->tm0.min_filter = FilterMode::Linear;
-    sampler->tm0.mag_filter = FilterMode::Linear;
-    sampler->tm0.mipmap_filter = FilterMode::Linear;
-  }
-  else if (sampler_state_filter == "point")
-  {
-    sampler->tm0.min_filter = FilterMode::Linear;
-    sampler->tm0.mag_filter = FilterMode::Linear;
-    sampler->tm0.mipmap_filter = FilterMode::Linear;
+      if (uv_mode == "clamp")
+      {
+        if (uv == "u")
+        {
+          sampler->tm0.wrap_u = WrapMode::Clamp;
+        }
+        else
+        {
+          sampler->tm0.wrap_v = WrapMode::Clamp;
+        }
+      }
+      else if (uv_mode == "repeat")
+      {
+        if (uv == "u")
+        {
+          sampler->tm0.wrap_u = WrapMode::Repeat;
+        }
+        else
+        {
+          sampler->tm0.wrap_v = WrapMode::Repeat;
+        }
+      }
+      else if (uv_mode == "mirror")
+      {
+        if (uv == "u")
+        {
+          sampler->tm0.wrap_u = WrapMode::Mirror;
+        }
+        else
+        {
+          sampler->tm0.wrap_v = WrapMode::Mirror;
+        }
+      }
```

Comment:
I feel like this might be a bit nicer if you first convert the string into a `WrapMode` and then assign it, so you only need one uv == u / v check instead of three.

Maybe even make it into a function, and then you can remove the for loop too, and just
```cpp
if (std::optional<WrapMode> u = ReadWrapModeFromJSON(sampler_state_wrap_obj, "u"))
    sampler->tm0.wrap_u = *u;
else
    return false;
if (std::optional<WrapMode> v = ReadWrapModeFromJSON(sampler_state_wrap_obj, "v"))
    sampler->tm0.wrap_v = *v;
else
    return false;
```

Same for the filter below